subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/TestPageGenerator.java

```diff
@@ -193,17 +189,57 @@ private PerformanceTestExecution findPreviousExecutionInSameBranch(PerformanceTe
                 return previousResults;
             }
 
-            private void addPerformanceGraph(String heading, String chartId, String jsonFieldName, String fieldLabel, String fieldUnit) {
-                h3().text(heading).end();
-                div().id(chartId).classAttr("chart");
-                p().text("Loading...").end();
+            private void addPerformanceGraphs() {
+                List<Chart> charts = Lists.newArrayList(new Chart("totalTime", "total time", "s", "totalTimeChart"));
+                if(testHistory instanceof CrossVersionPerformanceTestHistory) {
+                    charts.add(new Chart("confidence", "confidence", "%", "confidenceChart"));
+                    charts.add(new Chart("difference", "difference", "%", "differenceChart"));
+                }
+
+                charts.forEach(chart -> {
+                    h3().text(StringUtils.capitalize(chart.getLabel())).end();
```

Comment:
It would be nice to have `Confidence percentage` or `Difference percentage` as graph header. Or add the unit to the y axis.